irc/api.go

```diff
@@ -222,3 +228,179 @@ func (a *ergoAPI) handleAccountDetails(w http.ResponseWriter, r *http.Request) {
+	} else {
+		response.Success = false
+		response.ErrorCode = "INVALID_REQUEST"
+	}
+
+	a.writeJSONResponse(response, w, r)
+}
+
+type apiAccountListRequest struct {
+	Limit  int    `json:"limit,omitempty"`
+	Offset int    `json:"offset,omitempty"`
+	Filter string `json:"filter,omitempty"`
+}
+
+type apiAccountListResponse struct {
+	Accounts   []apiAccountDetailsResponse `json:"accounts"`
+	TotalCount int                         `json:"totalCount"`
+}
+
+func (a *ergoAPI) handleAccountList(w http.ResponseWriter, r *http.Request) {
+	var request struct {
+		Limit  int    `json:"limit,omitempty"`
+		Filter string `json:"filter,omitempty"`
+	}
+	if err := a.decodeJSONRequest(&request, w, r); err != nil {
+		return
+	}
+
+	var response apiAccountListResponse
+	var accounts []string
+
+	// Get all account names
+	accountNamePrefix := fmt.Sprintf(keyAccountName, "")
+	a.server.store.View(func(tx *buntdb.Tx) error {
+		return tx.AscendGreaterOrEqual("", accountNamePrefix, func(key, value string) bool {
+			if !strings.HasPrefix(key, accountNamePrefix) {
+				return false
+			}
+			accounts = append(accounts, value)
+			return true
+		})
+	})
+
+	response.TotalCount = len(accounts)
+
+	// Apply filtering if requested
+	if request.Filter != "" {
+		filtered := make([]string, 0, len(accounts))
+		cfFilter, _ := CasefoldName(request.Filter)
+		for _, account := range accounts {
+			cfAccount, _ := CasefoldName(account)
+			if strings.Contains(cfAccount, cfFilter) || strings.Contains(account, request.Filter) {
+				filtered = append(filtered, account)
+			}
+		}
+		accounts = filtered
+		response.TotalCount = len(accounts) // update total count after filter
+	}
+
+	// Apply limit if requested
+	if request.Limit > 0 && request.Limit < len(accounts) {
+		accounts = accounts[:request.Limit]
+	}
+
+	// Load account details
+	response.Accounts = make([]apiAccountDetailsResponse, len(accounts))
+	for i, account := range accounts {
+		accountData, err := a.server.accounts.LoadAccount(account)
+		if err != nil {
+			response.Accounts[i] = apiAccountDetailsResponse{
+				apiGenericResponse: apiGenericResponse{
+					Success: false,
+					Error:   err.Error(),
+				},
+			}
+			continue
+		}
+
+		response.Accounts[i] = apiAccountDetailsResponse{
+			apiGenericResponse: apiGenericResponse{
+				Success: true,
+			},
+			AccountName: accountData.Name,
+			Email:       accountData.Settings.Email,
+		}
+	}
+
+	a.writeJSONResponse(response, w, r)
+}
+
+
+
+
+type healthCheckResponse struct {
+	Version   string `json:"version"`
+	GoVersion string `json:"go_version"`
+	StartTime string `json:"start_time"`
+	Users     struct {
+		Total     int `json:"total"`
+		Invisible int `json:"invisible"`
+		Operators int `json:"operators"`
+		Unknown   int `json:"unknown"`
+		Max       int `json:"max"`
+	} `json:"users"`
+	Channels int `json:"channels"`
+	Servers  int `json:"servers"`
+}
+
+func (a *ergoAPI) handleHealthCheck(w http.ResponseWriter, r *http.Request) {
```

Comment:
I approve of the larger scope you chose for this endpoint but I think `/status` is actually a better name at this point.

BTW all responses should have a `success` boolean field.

Reply:
You can embed `apiGenericResponse` in your custom response object to get the success and error fields, e.g.

https://github.com/ergochat/ergo/blob/a1c46a4be75c27f09ae458d5e827ca94116a70a4/irc/api.go#L176-L180